service/controller/resource/instance/create.go

```diff
@@ -66,14 +62,19 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		}
 	}
 
+	err = r.CtrlClient.Status().Update(ctx, &azureMachinePool)
```

Comment:
We are saving the `Status` field of the CR. This is to be able to save the provisioning state.